examples/async_job_example.rb

```diff
@@ -0,0 +1,117 @@
+# Example of an async-compatible job for use with Gouda's fiber scheduler
+
+require "async"
+require "async/http"
+
+class AsyncHttpJob < ApplicationJob
+  queue_as :default
+
+  def perform(url)
+    # This job will work with both thread-based and fiber-based workers
+    # When using fiber scheduler, HTTP requests will be non-blocking
+
+    # Use async-compatible HTTP client
+    response = fetch_url_async(url)
+
+    # Process the response
+    process_response(response)
+  end
+
+  private
+
+  def fetch_url_async(url)
+    # When running under fiber scheduler, this will be non-blocking
+    # When running under thread scheduler, this will block the thread
```

Comment:
This is slightly misleading I think - if you do `Async do` but there is no top-level reactor already running, your `Async` call will be its own reactor, right? So both these calls will use Async, and both of them will be blocking - it's just that with a higher-level reactor present this call will be concurrent with other calls that use Async.

It's not nitpicking - we must make every effort to make Async setups easier to understand